code/game/objects/items/weapons/tether.dm

```diff
@@ -0,0 +1,88 @@
+var/list/global/all_tethers = list()
+
+/obj/item/tethering_device
+	name = "tethering device"
+	desc = "A device used by explorers to keep track of partners by way of electro-tether."
+	desc_info = "Use in-hand to activate, must be on the same level and within fifteen tiles of another device to latch. Tethers are colour coded by distance."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "locator"
+	w_class = ITEMSIZE_SMALL
+	slot_flags = SLOT_BELT
+	force = 1
+	var/active = FALSE
+	var/list/linked_tethers = list()
+	var/tether_range = 15
+	var/list/active_beams = list()
+
+/obj/item/tethering_device/Initialize(mapload, ...)
+	. = ..()
+	all_tethers += src
+
+/obj/item/tethering_device/Destroy()
+	STOP_PROCESSING(SSprocessing, src)
+	all_tethers -= src
+	deactivate()
+	return ..()
+
+/obj/item/tethering_device/attack_self(mob/user)
+	active = !active
+	var/msg = "You [active ? "activate" : "deactivate"] \the [src]."
+	to_chat(user, SPAN_NOTICE(msg))
+	if(active)
+		activate()
+	else
+		deactivate()
+
+/obj/item/tethering_device/process()
```

Comment:
Since these beams only care about their own positions for their calculations, I think it would be much better to register them to movement handlers, their own or their holder, and use that to trigger this process, instead of doing it every tick for every beam that is active. And also in attack_self() when you're activating or deactivating one of them.

Reply:
is there a massive benefit to doing that? i'm not opposed to it, but it seems a bit risky to not have them checking on their own. how expensive is this process?

Reply:
checked it out with handlers, it seemed to proc twice if i walked with it in my hands, and never when i walked with it in my backpack. seems more expensive to use as well, if eight people are going to be sprinting around with it somewhere.

Reply:
I don't know why it is proccing twice, possibly you registered to two different movement events somehow.

It can't be more expensive than process. If 8 people are running around with tethers, they're all calculating just the same anyway.